lib/collection/property-list.js

```diff
@@ -155,34 +160,35 @@ _.extend(PropertyList.prototype, /** @lends PropertyList.prototype */ {
      * @param context {Object} Optional context to bind the predicate to.
      */
     remove: function (predicate, context) {
-        var match; // to be used if predicate is an ID
+        var match, // to be used if predicate is an ID
+            list = this;
 
-        !context && (context = this);
+        !context && (context = list);
         // if predicate is id, then create a function to remove that from array
         if (_.isString(predicate)) {
-            (match = this.one(predicate)) && (predicate = function (item) {
+            (match = list.one(predicate)) && (predicate = function (item) {
```

Comment:
Like we discussed the other day, this pattern can cause potential memory leaks. While it's not better or worse than a `.bind()`, there is no reason to change this here.